mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1636,6 +1639,84 @@ private static ReadPreference getReadPreference(AssociationContext associationCo
 		return associationContext.getAssociationTypeContext().getOptionsContext().getUnique( ReadPreferenceOption.class );
 	}
 
+
+	@Override
+	public boolean supportsNamedPosition() {
+		return false;
+	}
+
+	/**
+	 * call stored procedure
+	 * @param storedProcedureName name of stored procedure.
+	 * @param params - array with values
+	 * @param tupleContext the tuple context
+	 *@see <a href="https://stackoverflow.com/questions/32480060/call-mongodb-function-from-java"> example</a>
+	 * @return
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public ClosableIterator<Tuple> callStoredProcedure(String storedProcedureName, Object[] params,
+			TupleContext tupleContext) {
+		StringBuffer commandLine = new StringBuffer( storedProcedureName ).append( "(" );
+		for ( Object param : params ) {
+			if ( param instanceof String ) {
+				commandLine.append( '\'' ).append( param.toString() ).append( '\'' );
+			}
+			else {
+				commandLine.append( param.toString() );
+			}
```

Comment:
I don't understand this code, in both cases you are converting the parameter to a String but only  in the first case you are adding quotes.

I would expect something like:

    if( hasToBeEscaped( param )) { ... }

Where` hasToBeEscaped` checks if the value has some invalid characters.

Reply:
OK, I get it now. This method seems too simple , I think you should check:

1) What happens if the String contains `'`. You need to escape other characters for sure.
2) What happens if the object is not a String but the result of the toString()  is actually a String (and not a Number for example)?

Reply:
Because strings must be escaped ... Okey ... good idea ..  I will correct my code.

Reply:
I added escaping for String parameters